File: apps/admin/src/api/types/variable.ts
/**
 * One variable, as one environment sees it (D57).
 *
 * The name and description belong to the **project** and read the same in every
 * environment; `value` and `set_in` are what change as you switch environment.
 *
 * `value` is `null` when this environment has given it nothing, which is not
 * the same as `''` — an empty value substitutes as empty, an unset one leaves
 * `{{NAME}}` standing in the API response. The page says which.
 */
export interface Variable {
  name: string
  description: string
  value: string | null
  /** How many of the project's environments have a value for it. */
  set_in: number
  created_at: string
  updated_at: string
}

File: packages/silo-client-dart/lib/src/variables/variable.dart
import '../transport/timestamps.dart';

/// One `{{NAME}}` as one environment sees it. [value] is null when this
/// environment has set nothing; `''` is a value.
final class Variable {
  const Variable({
    required this.name,
    required this.description,
    required this.value,
    required this.setIn,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Variable.fromJson(Map<String, Object?> json) => Variable(
    name: json['name'] as String,
    description: json['description'] as String,
    value: json['value'] as String?,
    setIn: json['set_in'] as int,
    createdAt: Timestamps.parse(json['created_at']),
    updatedAt: Timestamps.parse(json['updated_at']),
  );

  final String name;
  final String description;
  final String? value;
  final int setIn;
  final DateTime createdAt;
  final DateTime updatedAt;
}
